cpp/src/plasma/lib/java/org_apache_arrow_plasma_PlasmaClientJNI.cc

```diff
@@ -0,0 +1,314 @@
+  }
+
+  std::shared_ptr<Buffer> data;
+  Status s = client->Create(oid, size, md, md_size, &data);
+  if (s.IsPlasmaObjectExists()) {
+    jclass Exception = env->FindClass("org/ray/spi/impl/PlasmaObjectExistsException");
+    env->ThrowNew(Exception,
+                  "An object with this ID already exists in the plasma store.");
+    return NULL;
+  }
+  if (s.IsPlasmaStoreFull()) {
+    jclass Exception = env->FindClass("org/ray/spi/impl/PlasmaOutOfMemoryException");
+    env->ThrowNew(Exception,
+                  "The plasma store ran out of memory and could not create this object.");
+    return NULL;
+  }
+  ARROW_CHECK(s.ok());
+
+  return env->NewDirectByteBuffer(data->mutable_data(), size);
+}
+
+JNIEXPORT jbyteArray JNICALL Java_org_apache_arrow_plasma_PlasmaClientJNI_hash(
+    JNIEnv* env, jclass cls, jlong conn, jbyteArray object_id) {
+  plasma::PlasmaClient* client = (plasma::PlasmaClient*)conn;
+  plasma::ObjectID oid;
+  jbyteArray_to_object_id(env, object_id, &oid);
+
+  unsigned char digest[plasma::kDigestSize];
+  bool success = client->Hash(oid, digest).ok();
+
+  if (success) {
+    jbyteArray ret = env->NewByteArray(plasma::kDigestSize);
+    env->SetByteArrayRegion(ret, 0, plasma::kDigestSize,
+                            reinterpret_cast<jbyte*>(digest));
+    return ret;
+  } else {
+    return NULL;
+  }
+}
+
+JNIEXPORT void JNICALL Java_org_apache_arrow_plasma_PlasmaClientJNI_seal(
+    JNIEnv* env, jclass cls, jlong conn, jbyteArray object_id) {
+  plasma::PlasmaClient* client = (plasma::PlasmaClient*)conn;
+  plasma::ObjectID oid;
+  jbyteArray_to_object_id(env, object_id, &oid);
+
+  ARROW_CHECK_OK(client->Seal(oid));
+}
+
+JNIEXPORT void JNICALL Java_org_apache_arrow_plasma_PlasmaClientJNI_release(
+    JNIEnv* env, jclass cls, jlong conn, jbyteArray object_id) {
+  plasma::PlasmaClient* client = (plasma::PlasmaClient*)conn;
+  plasma::ObjectID oid;
+  jbyteArray_to_object_id(env, object_id, &oid);
+
+  ARROW_CHECK_OK(client->Release(oid));
+}
+
+JNIEXPORT jobjectArray JNICALL Java_org_apache_arrow_plasma_PlasmaClientJNI_get(
+    JNIEnv* env, jclass cls, jlong conn, jobjectArray object_ids, jint timeout_ms) {
+  plasma::PlasmaClient* client = (plasma::PlasmaClient*)conn;
+
+  jsize num_oids = env->GetArrayLength(object_ids);
+  std::vector<plasma::ObjectID> oids(num_oids);
+  std::vector<plasma::ObjectBuffer> obufs(num_oids);
+  for (int i = 0; i < num_oids; ++i) {
+    jbyteArray_to_object_id(env, (jbyteArray)env->GetObjectArrayElement(object_ids, i),
+                            &oids[i]);
+  }
+  // TODO: may be blocked. consider to add the thread support
+  ARROW_CHECK_OK(client->Get(oids.data(), num_oids, timeout_ms, obufs.data()));
+
+  jclass clsByteBuffer = env->FindClass("java/nio/ByteBuffer");
+  jclass clsByteBufferArray = env->FindClass("[Ljava/nio/ByteBuffer;");
+
+  jobjectArray ret = env->NewObjectArray(num_oids, clsByteBufferArray, NULL);
+  jobjectArray o = NULL;
```

Comment:
Should these `NULL`s be `nullptr`? I'm not sure.

Reply:
It is point here, maybe nullptr better, i will fix them.